src/gui/SubWindow.cpp

```diff
@@ -158,21 +166,33 @@ void SubWindow::changeEvent( QEvent *event )
 	{
 		adjustTitleBar();
 	}
-
 }
 
+
+
+
 void SubWindow::setVisible(bool visible)
 {
-	if (isDetached() && visible)  // avoid showing titlebar here
+	if (isDetached() || visible) { widget()->setVisible(visible); }
+	if (!isDetached()) { QMdiSubWindow::setVisible(visible); }
```

Comment:
`widget->setVisible(true)` for attached widgets is ensured by `showEvent`

```suggestion
	if (isDetached()) { widget()->setVisible(visible); }
	else { QMdiSubWindow::setVisible(visible); }
```

Reply:
This is/was a fix actually, without the `|| visible` part, windows that are initialized are never shown before the `hide()` is called drop their height for some reason. Easiest case to test is editor windows - they just appear as toolbars only initially (btw, they probably should have a minimal height higher than that)

Edit: Though I guess under normal circumstances you can't detach a hidden window and try to hide it again, but I don't like such implicit logic. Maybe I should remove the `show()` from `showEvent` instead?
\

Reply:
This seems to happen when a MDI window opens for the first time. It will try to calculate its size, and if the widget inside is hidden the size will be 0. If the widget is then shown it will expand the window to the minimum size of the widget (which is the toolbar in case of piano roll).

You can leave it like that for now and remove the show from showEvent.